### PR TITLE
Address some FIXMEs for ra_assists

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### CODE REVIEW

**Comment by JmPotato** [comment]
bors try

File: crates/ra_assists/Cargo.toml [hunk 1]
```diff
@@ -22,4 +22,5 @@ ra_prof = { path = "../ra_prof" }
 ra_db = { path = "../ra_db" }
 ra_ide_db = { path = "../ra_ide_db" }
 hir = { path = "../ra_hir", package = "ra_hir" }
+hir_expand = { path = "../ra_hir_expand", package = "ra_hir_expand" }
```
Comment (lnicola) [inline crates/ra_assists/Cargo.toml:25]: I think the idea is to export everything from `ra_hir` and only depend on that crate.

Comment (JmPotato) [reply crates/ra_assists/Cargo.toml]: Thanks for pointing out. Already fixed!

**Comment by lnicola** [comment]
The typo fixes and visibility change are fine, but I'm not sure about the hygiene. It might be better if the code used `Path::from_src` instead. Maybe it's possible to construct a `Hygiene` instance?

**Comment by lnicola** [comment]
CC https://github.com/rust-analyzer/rust-analyzer/pull/5708

**Comment by JmPotato** [comment]
> The typo fixes and visibility change are fine, but I'm not sure about the hygiene. It might be better if the code used `Path::from_src` instead. Maybe it's possible to construct a `Hygiene` instance?

You are right. Using `Path::from_src` but without a `Hygiene` instance will make this change meaningless. I'll do some work to figure out and fix this. BTW, could you please tell me what's the use of `Hygiene` from a higher perspective? I'm kind of a rust-analyzer noob here...

**Comment by lnicola** [comment]
I'm not sure myself, but I think it's used for macros (which get expanded into a virtual file). Check out the docstring of `HirFileId`.

**Comment by JmPotato** [comment]
> I'm not sure myself, but I think it's used for macros (which get expanded into a virtual file). Check out the docstring of `HirFileId`.

I saw the explanation in comments and checked #5708. Now I'm confused about how to get a `HirFileId` in this case. Do you have any suggestions or ideas? Thanks a lot!

**Comment by lnicola** [comment]
Did you see https://github.com/rust-analyzer/rust-analyzer/pull/5708? If you have a `FileId`, you can call `.into()` on it.

**Comment by lnicola** [comment]
Ah, you don't have one, do you? I'm not sure then..

**Comment by JmPotato** [comment]
> Ah, you don't have one, do you? I'm not sure then..

It seems that `FileId` doesn't even be involved in this part.

**Comment by lnicola** [comment]
I guess you could pass it to `SubstituteTypeParams` (it's available in `Assists`, although private), but I'm not sure if it makes sense.

**Comment by JmPotato** [comment]
> I guess you could pass it to `SubstituteTypeParams` (it's available in `Assists` and `AssistBuilder`, although private), but I'm not sure if it makes sense.

I don't think it's a good idea...Maybe I should revert changes about `Path::from_src` and get the FIXMEs back first unless we find a good way to handle it well.

**Comment by lnicola** [comment]
> I don't think it's a good idea...Maybe I should revert changes about Path::from_src and get the FIXMEs back first unless we find a way to handle it well.

I think that will be the solution in the end, but we can postpone this until we're sure.

**Comment by JmPotato** [comment]
> > I don't think it's a good idea...Maybe I should revert changes about Path::from_src and get the FIXMEs back first unless we find a way to handle it well.
> 
> I think that will be the solution in the end, but we can postpone this until we're sure.

I'll revert changes about `Path::from_src` which I will open a new PR to fix later. Let's keep other changes go first. Thanks a lot for your patient help! :-)

**Pull request merged**